File: internal/content/testdata/bench/csharp/nc-combination-sum-ii.cs
using System.Collections.Generic;

public class Solution {
    public long[][] combination_sum2(long[] candidates, long target) {
        var res = new List<List<long>>();
        var sorted = (long[])candidates.Clone();
        System.Array.Sort(sorted);

        void Backtrack(int start, List<long> current, long remaining) {
            if (remaining == 0) {
                res.Add(new List<long>(current));
                return;
            }
            for (int i = start; i < sorted.Length; i++) {
                if (sorted[i] > remaining) break;
                if (i > start && sorted[i] == sorted[i - 1]) continue;
                current.Add(sorted[i]);
                Backtrack(i + 1, current, remaining - sorted[i]);
                current.RemoveAt(current.Count - 1);
            }
        }

        Backtrack(0, new List<long>(), target);

        foreach (var c in res) c.Sort();
        res.Sort((a, b) => {
            int len = Math.Min(a.Count, b.Count);
            for (int i = 0; i < len; i++) {
                int cmp = a[i].CompareTo(b[i]);
                if (cmp != 0) return cmp;
            }
            return a.Count.CompareTo(b.Count);
        });

        return res.ConvertAll(c => c.ToArray()).ToArray();
    }
}

File: internal/content/testdata/bench/javascript/nc-combination-sum-ii.js
function combination_sum2(candidates, target) {
    const res = [];
    candidates.sort((a, b) => a - b);
    function backtrack(start, current, remaining) {
        if (remaining === 0) {
            res.push([...current]);
            return;
        }
        for (let i = start; i < candidates.length; i++) {
            if (candidates[i] > remaining) break;
            if (i > start && candidates[i] === candidates[i - 1]) continue;
            current.push(candidates[i]);
            backtrack(i + 1, current, remaining - candidates[i]);
            current.pop();
        }
    }
    backtrack(0, [], target);
    for (let i = 0; i < res.length; i++) res[i].sort((a, b) => a - b);
    res.sort((a, b) => {
        for (let i = 0; i < Math.min(a.length, b.length); i++) {
            if (a[i] !== b[i]) return a[i] - b[i];
        }
        return a.length - b.length;
    });
    return res;
}
